Creates AI Assistant use cases section (#5303)
elastic/security-docs: docs/assistant/ai-esql-queries.asciidoc

[[esql-queries-assistant]]
= Generate, customize, and learn about {esql} queries

:frontmatter-description: Elastic AI Assistant can help you write ES|QL queries.
:frontmatter-tags-products: [security]
:frontmatter-tags-content-type: [guide]
:frontmatter-tags-user-goals: [get-started]

> This page describes some of the ways Elastic AI Assistant can help you leverage the Elasticsearch Query Language ({esql}). 

Reviewer: It feels a little unnecessary for the page to announce itself in this way. (I know that works in other situations, so I'm not sure why it doesn't work for me here. Maybe because it's such a simple, direct statement, and the extra phrasing dilutes its effect?)
```suggestion
Elastic AI Assistant can help you learn about and leverage the Elasticsearch Query Language ({esql}). 
```

With AI Assistant's <<ai-assistant-knowledge-base, {esql} knowledge base>> enabled, AI Assistant benefits from specialized training data that enables it to answer questions related to {esql} at an expert level. 

AI Assistant can help with {esql} in many ways, including:

> * **Education and training**: AI Assistant can serve as a powerful {esql} learning tool. Ask it for examples, explanations of complex queries, and best practices.

Reviewer 2: ```suggestion
* **Education and training**: AI Assistant can be a powerful {esql} learning tool. Ask it for examples, explanations of complex queries, and best practices.
```

> * **Writing new queries**: Prompt AI Assistant to provide a query that accomplishes a particular task, and it will generate a query matching your description. For example: "Write a query to identify documents with `curl.exe` usage and calculate the sum of `destination.bytes`" or "What query would return all user logins to [a host] in the last six hours?"

Reviewer 2: ```suggestion
* **Writing new queries**: Prompt AI Assistant to provide a query that accomplishes a particular task, and it will generate a query that matches your description. For example: "Write a query to identify documents with `curl.exe` usage and calculate the sum of `destination.bytes`" or "What query would return all user logins to [a host] in the last six hours?"
```

> * **Providing feedback to optimize existing queries**: Send AI Assistant a query you want to work on and ask it for improvements, refactoring, a general assessment, or to optimize the query's performance with large data sets.

Reviewer 2: ```suggestion
* **Providing feedback to optimize existing queries**: Provide AI Assistant a query you want to use and ask it for improvements, refactoring, a general assessment, or to optimize the query's performance with large data sets.
```

* **Customizing queries for your environment**: Since each environment is unique, you may need to customize queries that you used in other contexts. AI Assistant can suggest necessary modifications based on contextual information you provide. 
* **Troubleshooting**: Having trouble with a query or getting unexpected results? Ask AI Assistant to help you troubleshoot.

> In these ways and others, AI Assistant can enable you to make use of {esql}'s advanced search capabilities to accomplish goals across {elastic-sec}. 

Reviewer 2: ```suggestion
In these ways and others, AI Assistant can enable you to make use of {esql}'s advanced search capabilities to accomplish goals within {elastic-sec}. 
```

[discrete]
== Related documentation

> Get started with <<security-assistant, AI Assistant>>, or learn more about {ref}/esql-kibana.html[{esql}].

Reviewer: Hm, I really like how this "links richly" to build an EPPO-style information web 😉, but I'm not sure about the format. It's a super short section with just one sentence, which feels out of place (which don't typically have one-sentence sections in general). It might also feel more natural if ALL our pages all included a "Related docs" section, but it might not work as a one-off.

What about embedding these links into the intro paragraph instead? It already references "Elastic AI Assistant" and "Elasticsearch Query Language" so those would be good anchors.
